### DISCUSSION TITLE
Failed to start gauge api timeout connecting to java in AWS fargate cluster

### DISCUSSION THREAD

**Discussion by indoriautomation** [getgauge/gauge#2213] [General]
Getting gauge api timeout connecting to java exception in AWS fargate cluster build with : Amazon Linux 2 OS, version :"4.14.262-200.489.amzn2.x86_64 ", arch:"amd64" 
Gauge version : 1.4.3
Plugins Versions:- Gauge-java(0.7.15), Screenshot(0.1.0), xml-report(0.2.3), Html-report(4,1,4),Flash(0.0.2)
Open JDK:- Java 1.8.0_275
gauge config ide_request_timeout 600000 , plugin_connection_timeout 600000, plugin_kill_timeout 300000, runner_connection_timeout 600000, runner_request_timeout 600000
gauge-maven-plugin:1.5.0

Any idea or suggestion what's wrong in there that causing gauge timeout issue?

**Comment by zabil** [comment]
You can try the following
* Upgrade Java to 11+
* Increase the java memory settings to have atleast 1gb

**Reply by indoriautomation** [reply]
Thanks for your response, I tried with suggested Java upgraded version (OpenJDK 11.0.5) and having memory setting more than 2 GB(maxheapsize), still getting same error :--
Failed to start gauge api : Timeout connecting to java.

**Reply by zabil** [reply]
Can you share more info on how you set the java properties? Have you set this values in the [`java.properties`](https://docs.gauge.org/writing-specifications.html?os=macos&language=java&ide=vscode#language-specific-project-files) file?

Can you also share a bit more info which can be found in relevant sections of log files available under the `logs` directory of your project.

**Reply by indoriautomation** [reply]
Just to update on this thread : When all gauge timeout set to more than 30 min, its worked but gauge kick of test execution after 30 min only. While same solution working fine on local windows and linux infrastructure and invoke gauge within 1 min.

**Reply by indoriautomation** [reply]
Yes setting java memory in the java.properties gauge_jvm_args= -xms2048, -xmx2048

**Reply by indoriautomation** [reply]
Any suggestion what would be possible causes behind gauge api started executing test after 30 min in aws alpine 3.11 container with openJDK11 and how that can be reduce to 1 min or so to make test execution initiation faster?  In logs folder, gauge.log file is created that to have gauge timeout message only..

**Comment by sriv** [comment]
https://github.com/getgauge/gauge/discussions/2214 seems to be a similar problem but on a diff env. I suspect there is a pattern here.

/cc @Laurie-Bamber
